From 25ca70e4dbf33dff1ab0c7c8ede92093ee1f59f2 Mon Sep 17 00:00:00 2001
From: Robert Trace <gentoobugs@farcaster.org>
Date: Tue, 23 Feb 2010 19:44:56 +0100
Subject: [PATCH] Clear archive_error_number in archive_clear_error()

From http://bugs.gentoo.org/show_bug.cgi?id=289260#c1 :

  archive_clear_error() doesn't appear to be clearing all of the internally
  stored error fields.  Since archive_error_number is exposed via the
  archive_errno() interface (and gvfs, at least, is consuming this data), this
  could lead to false failures.

This fixes a crash in gvfsd-archive.
---
 libarchive/archive_util.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/libarchive/archive_util.c b/libarchive/archive_util.c
index c945d5f..c0104d6 100644
--- a/libarchive/archive_util.c
+++ b/libarchive/archive_util.c
@@ -153,6 +153,7 @@ archive_position_uncompressed(struct archive *a)
 void
 archive_clear_error(struct archive *a)
 {
+	a->archive_error_number = 0;
 	archive_string_empty(&a->error_string);
 	a->error = NULL;
 }
-- 
1.7.0

